guard/src/rules/functions/converters.rs

```diff
@@ -0,0 +1,214 @@
+use crate::rules::{
+    path_value::{Path, PathAwareValue},
+    QueryResult,
+};
+
+pub(crate) fn parse_float(
+    args: &[QueryResult],
+) -> crate::rules::Result<Vec<Option<PathAwareValue>>> {
+    let mut aggr = vec![];
+    for entry in args.iter() {
+        match entry {
+            QueryResult::Literal(val) | QueryResult::Resolved(val) => match &**val {
+                PathAwareValue::String((path, val)) => {
+                    let number = match val.parse::<f64>() {
+                        Ok(f) => Some(PathAwareValue::Float((path.clone(), f))),
+                        Err(_) => {
+                            return Err(crate::Error::ParseError(format!(
+                                "attempting to convert a string: {val} into a number at {path}"
+                            )))
+                        }
+                    };
+
+                    aggr.push(number)
+                }
+                PathAwareValue::Int((path, val)) => {
+                    aggr.push(Some(PathAwareValue::Float((path.clone(), *val as f64))))
+                }
+                PathAwareValue::Float((path, val)) => {
+                    aggr.push(Some(PathAwareValue::Float((path.clone(), *val))))
+                }
+                PathAwareValue::Char((path, val)) => aggr.push(Some(PathAwareValue::Float((
+                    path.clone(),
+                    convert_char_to_number(val, path)? as f64,
+                )))),
+                _ => {
+                    aggr.push(None);
+                }
+            },
+            _ => {
+                aggr.push(None);
+            }
+        }
+    }
+
+    Ok(aggr)
+}
+
+fn convert_char_to_number(val: &char, path: &Path) -> crate::rules::Result<u32> {
+    val.to_digit(10).ok_or(crate::Error::ParseError(format!(
+        "attempting to convert a string: {val} into a number at {path}"
```

Comment:
the function name is char to number but error message is to convert a string, is this expected?

Reply:
good catch.